api/src/models/request_info.rs

```diff
@@ -67,16 +73,40 @@ impl<'r> FromRequest<'r> for RequestInfo {
     type Error = ();
 
     async fn from_request(request: &'r Request<'_>) -> request::Outcome<Self, Self::Error> {
-        let hosts: Vec<_> = request.headers().get("host").collect();
+        // Checking if "x-forwarded-host" header is present
+        let forwarded_host = request.headers().get_one("x-forwarded-host");
+        let host = match forwarded_host {
+            Some(proto) => proto.to_string(),
+            None => request.headers().get_one("host").unwrap().to_string(), // Default to "host" if not present
+        };
 
-        if hosts.len() != 1 {
+        if host == "" {
             return Outcome::Failure((Status::BadRequest, ()));
         }
 
-        let url_string = format!("http://{}", hosts[0]);
-        match Url::parse(&url_string) {
-            Ok(url) => return Outcome::Success(RequestInfo { base_url: url }),
-            Err(_) => return Outcome::Failure((Status::BadRequest, ())),
+        // Checking if "x-forwarded-proto" header is present
+        let forwarded_proto_header = request.headers().get_one("x-forwarded-proto");
+        let forwarded_proto = match forwarded_proto_header {
+            Some(proto) => proto.to_string(),
+            None => "http".to_string(), // Default to "http" if not present
+        };
```

Comment:
I would rewrite this and the match below and use would use [`unwrap_or_else`](https://api.rocket.rs/v0.5-rc/rocket/mtls/oid/asn1_rs/nom/lib/std/option/enum.Option.html#method.unwrap_or_else) because that makes these `// default to … if not present` comments obsolete.

```suggestion
        let forwarded_proto = request.headers().get_one("x-forwarded-proto").map(|proto| proto.to_string()).unwrap_or_else(|| String::from("http"));
```